.github/workflows/staterecovery-testing.yml

```diff
@@ -0,0 +1,90 @@
+name: staterecovery-testing
+
+on:
+  workflow_call:
+    inputs:
+      commit_tag:
+        required: true
+        type: string
+    secrets:
+      DOCKERHUB_USERNAME:
+        required: false
+      DOCKERHUB_TOKEN:
+        required: false
+
+concurrency:
+  group: staterecovery-testing-${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+jobs:
+  cache-docker-images:
+    uses: ./.github/workflows/cache-docker-images.yml
+    secrets: inherit
+  run-tests:
+    env:
+      COMMIT_TAG: ${{ inputs.commit_tag }}
+      GITHUB_TOKEN: ${{ secrets._GITHUB_TOKEN_RELEASE_ACCESS }}
+      DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+      DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+      CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+    runs-on: gha-runner-scale-set-ubuntu-22.04-amd64-large
+    name: Coordinator tests
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - uses: actions/setup-java@8df1039502a15bceb9433410b1a100fbe190c53b #v4.5.0
+        with:
+          distribution: temurin
+          java-version: 21
+      - name: Setup Gradle
+        # Configure Gradle for optimal use in GiHub Actions, including caching of downloaded dependencies.
+        # See: https://github.com/gradle/actions/blob/main/setup-gradle/README.md
+        uses: gradle/actions/setup-gradle@cc4fc85e6b35bafd578d5ffbc76a5518407e1af0 #v4.2.1
+      - name: Restore cached images
+        id: restore-cached-images
+        uses: actions/cache/restore@v4.0.2
+        with:
+          path: ~/docker-images
+          key: cached-images
+          restore-keys: |
+            cached-images
+      - name: Staterecovery - Build and Unit tests
+        run: |
+          ./gradlew state-recovery:besu-plugin:buildNeeded
+      - name: Login to Docker Hub
```

Comment:
Do we really need this login step here? maybe @kyzooghost or @VGau know best, If not, we could remove the dockerhub secrets